design/conjur-safe.md

```diff
@@ -0,0 +1,29 @@
+## Conjur Safe Resource
+
+A Safe is a higher order resource. It includes a number of variables, groups, and permissions to simplify the process of working with collections of variables.
+
+
+### Example
+The following is an example policy creating a Safe resource:
+```yml
+# policies/databases/foo.yml
+- !safe
+  id: foo-database
+  variables:
+    - url
+    - port
+    - username
+    - password
+```
+When a Safe resource is created, the following resource primitives are created:
```

Comment:
It might be worth adding here some thoughts on how we'll implement this - it seems to me that there are a wide range of possibilities.

I think of a safe (as you described it above) as essentially analogous to this policy snippet:

```yml
- !policy
  id: foo-database
  body:
    - &variables
      - !variable url
      - !variable port
      - !variable username
      - !variable password

    - !group readers
    - !group writers
    - !group admins

    # readers can read and execute
    - !permit
      resource: *variables
      privileges: [ read, execute ]
      role: !group readers

    # writers can update
    - !permit
      resource: *variables
      privileges: [ update ]
      role: !group writers

    # admins can read, execute, and update
    - !permit
      resource: *variables
      privileges: [ read, execute, update ]
      role: !group admins
```